solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -472,33 +461,20 @@ library DKG {
             "result under challenge is different than the submitted one"
         );
 
-        // Compute the actual group members hash by selecting actual members IDs
-        // based on seed used for current DKG execution.
-        bytes32 actualGroupMembersHash = keccak256(
-            abi.encodePacked(
-                self.sortitionPool.selectGroup(groupSize, bytes32(self.seed))
-            )
-        );
+        bool areSignaturesValid = hasValidSignatures(self, result);
+        bool areMembersValid = true;
+        if (areSignaturesValid) {
+            areMembersValid = hasValidGroupMembers(self, result);
+        }
```

Comment:
We talked with @tomaszslabon that it would be good to wrap this part into a try-catch in case the signature validation fails for whatever reason. In `submitDkgResult` we only validate length, and no that the signature has correct format, for example, if it is prefixed with `\x19Ethereum signed message:\n`

Reply:
Try/catch works only for failed external calls and contract creation calls (source: https://docs.soliditylang.org/en/latest/control-structures.html#try-catch)

Reply:
Yep, this will get expensive. We need to check how `hash.toEthSignedMessageHash().recover(...)` behaves. If the behavior is well-defined and we can protect against garbage data we might be able to avoid it.

Reply:
Extracted to a separate contract.